git.md

```diff
@@ -75,3 +75,41 @@ when you want to pull changes from `master` into your branch.
 Further reading:
 
 - [Advanced pull-request crafting](https://codeinthehole.com/tips/advanced-pull-request-crafting/)
+
+
+## Use "fix-up" commits to address code review comments
+
+Before a branch merges into master, the commits should be logically
+self-contained (see [Commit granularity](https://github.com/octoenergy/conventions/blob/master/git.md#commit-granularity) above). 
+
+But after a PR has been submitted for code review, there will often be feedback
+that needs implementing. One approach is to make the appropriate changes and
+then rebase these into the original commits which preserves the commit granularity requirement.
+
+However, this isn't so great for the reviewer who needs to re-review every
+commit once code-review is requested again.
+
+It's better to use temporary "fix-up" commits that the reviewer can easily
+evaluate to see if they address the comments. Once approved, these can easily be
+rebased into the PR branch before merge, achieving the same commit granularity
+requirements but making better use of the code reviewer's time.
+
+These have a special commit subject synxtax
+that git recognises and knows how to rebase them 
```

Comment:
"that git recognises and uses to orchestrate rebases performed with `--autosquash`."